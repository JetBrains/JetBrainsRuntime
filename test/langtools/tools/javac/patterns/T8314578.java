/**
 * @test /nodynamiccopyright/
 * @bug 8314578
 * @enablePreview
 * @summary Parsing of erroneous patterns succeeds
 * @compile/fail/ref=T8314578.out -XDrawDiagnostics T8314578.java
 */
public class T8314578 {
    record R1() {}
    record R2() {}

    static void test(Object o) {
        switch (o) {
            case R1() when o instanceof String s:
            case R2() when o instanceof Integer i:
                System.out.println("hello: " + i);
                break;
            default:
                break;
        }
    }

    static void test2(Object o) {
        switch (o) {
            case R1() when o instanceof String s:
                System.out.println("hello: " + s);
            case R2() when o instanceof Integer i:
                System.out.println("hello: " + i);
                break;
            default:
                break;
        }
    }

    static int unnamedInGuardsOK(String s) {
        return switch (s) {
            case String _ when s instanceof String _ ->  // should be OK
                    1;
            default ->
                    -1;
        };
    }
}