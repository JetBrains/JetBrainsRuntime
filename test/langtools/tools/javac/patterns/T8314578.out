T8314578.java:14:18: compiler.err.flows.through.from.pattern
T8314578.java:15:18: compiler.err.flows.through.to.pattern
T8314578.java:27:18: compiler.err.flows.through.to.pattern
- compiler.note.preview.filename: T8314578.java, DEFAULT
- compiler.note.preview.recompile
3 errors